src/main.rs

```diff
@@ -442,9 +451,25 @@ fn exec_on_packages(
             line.arg("--target");
             line.arg(target);
         }
-        packages
-            .iter()
-            .try_for_each(|pkg| exec_on_package(cx, pkg.id, &pkg.kind, &line, progress, keep_going))
+
+        if cx.parallel {
+            packages.par_iter().try_for_each(|pkg| {
+                exec_on_package(cx, pkg.id, &pkg.kind, &line, progress, keep_going)
+            })
+        } else {
+            packages.iter().try_for_each(|pkg| {
+                exec_on_package(cx, pkg.id, &pkg.kind, &line, progress, keep_going)
+            })
+        }
+    } else if cx.parallel {
+        cx.target.par_iter().try_for_each(|target| {
+            let mut line = line.clone();
+            line.arg("--target");
+            line.arg(target);
+            packages.iter().try_for_each(|pkg| {
+                exec_on_package(cx, pkg.id, &pkg.kind, &line, progress, keep_going)
+            })
+        })
```

Comment:
Minor nit: I think it might be better to nest this in a larger `else` clause that catches the entire `!target.is_empty()` case, rather than three individual cases of empty, non-empty parallel, non-empty non-parallel.